packages/blockchain-wallet-v4-frontend/src/data/components/buySell/sagas.ts

```diff
@@ -1390,27 +1385,27 @@ export default ({ api, coreSagas, networks }: { api: APIType; coreSagas: any; ne
         S.getFiatCurrency(yield select()) ||
         (yield select(selectors.core.settings.getCurrency)).getOrElse('USD')
 
-      const userSDDTierR = S.getUserSddEligibleTier(yield select())
-      if (!Remote.Success.is(userSDDTierR)) {
-        yield call(fetchSDDEligible)
-      }
       const state = yield select()
       const currentUserTier = selectors.modules.profile
         .getCurrentTier(state)
         .getOrFail('User has no tier')
-      const userSDDEligibleTier = S.getUserSddEligibleTier(state).getOrElse(1)
-      // only fetch non-eligible payment methods if user is not tier 2
-      const includeNonEligibleMethods = currentUserTier === 2
-      // if user is SDD tier 3 eligible, fetch limits for tier 3
-      // else let endpoint return default current tier limits for current tier of user
-      // double check if user is tier 2 and in case user is ignore this property
-      const includeTierLimits =
-        userSDDEligibleTier === SDD_TIER && currentUserTier !== 2 ? SDD_TIER : undefined
+      const userDataR = selectors.modules.profile.getUserData(state)
+
+      // Present all possible (but necessarily eligible) payment methods to SDD users
+      const includeEligibleOnlyPaymentMethods = currentUserTier === 2
+
+      /**
+       * If it is SDD flow, request limits for SDD.
+       * Otherwise, allow endpoint to determine limits for the current user tier.
+       */
+      const includeTierLimits = userDataR
+        .map((userData) => (SddFlow.isSddUser(userData) ? SddFlow.TIER : undefined))
```

Comment:
@milan-bc, that does not make sense, right?

Previously, we have been looking at whether user is eligible for T3 and in such case we have been requesting limits for T3 (assuming user is T1 at the time). Now, this check if user is T3 and then requests T3, otherwise omits this params allowing endpoint to determine the limits. So, it seems we just can fully omit this `includeTierLimits`, and we will be getting T3 limits for T3 users.

Reply:
So we are on the same page that this param can be completely omitted, right?